salt/pillar/http_json.py

```diff
@@ -52,16 +56,26 @@ def __virtual__():
     return True
 
 
-def ext_pillar(minion_id, pillar, url, with_grains=False):  # pylint: disable=W0613
+def ext_pillar(
+    minion_id, pillar, url, with_grains=False, **kwargs
```

Comment:
why are we accepting all kwargs here? Would like to avoid that if possible.

Reply:
yes, going to alter the original PR to focus on `header_dict`, which is the kwarg requested in the original issue